bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONParser.java

```diff
@@ -896,6 +921,27 @@ private void processNonStringValue(ValueType type) throws JsonParserException {
             }
         }
 
+        private void setValueToJsonType(ValueType type, Object value) {
+            switch (type) {
+                case ARRAY_ELEMENT:
+                    ((ArrayValue) this.currentJsonNode).append(value);
+                    break;
+                case FIELD:
+                    ((MapValueImpl<BString, Object>) this.currentJsonNode).put(
+                            StringUtils.fromString(this.fieldNames.pop()), value);
+                    break;
+                case VALUE:
```

Comment:
Can't we make this the default case and remove the one below?

Reply:
Changed